### PR TITLE
Add unit test for 7150 fix

### PR DESCRIPTION
**Pull request by congqixia**
Add Unit test cases for 7151 fix (#7272)

Related issue: #7150 

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/meta_snapshot.go [hunk 1]
```diff
@@ -104,9 +105,18 @@ func (ms *metaSnapshot) loadTs() error {
...
 		}
+		if curVer == version {
+			log.Debug("snapshot found save version with different revision", zap.Int64("revision", revision), zap.Int64("version", version))
+		}
 		strTs := string(resp.Kvs[0].Value)
+		if strTs == "0" {
+			//#issue 7150, index building inserted "0", skipping
+			//this is a special fix for backward compatibility, previous version will put 0 ts into snapshot building index
+			continue
+		}
```
Comment (congqixia) [inline internal/rootcoord/meta_snapshot.go:119]: This part filters “0” value ts from older version. It's just for backward compatibility. Shall we keep this snippet only for handling data error in RC period? @sunby @xiaofan-luan @czs007 @scsven

Comment (czs007) [reply internal/rootcoord/meta_snapshot.go:119]: I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

Comment (xiaofan-luan) [reply internal/rootcoord/meta_snapshot.go:119]: > I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

This suggestion looks good to me

Comment (congqixia) [reply internal/rootcoord/meta_snapshot.go:119]: > I suggest keeping this code, but adding some conversion logic to change 0ts to non-0ts storage. Delete this code when the GA version is released.

I don't think there is a way to modify etcd kv with specified revision. Maybe we needs to change the implementation of `SnapshotKV`.

**Pull request merged**